packages/diagrams/frontend/sirius-components-diagrams-reactflow/src/renderer/DiagramEditor.tsx

```diff
@@ -0,0 +1,71 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+
+import { useEffect, useState } from 'react';
+import { DiagramEditorProps, DiagramEditorState } from './DiagramEditor.types';
+import { DiagramRenderer } from './DiagramRenderer';
+import { cleanLayoutArea, layout, prepareLayoutArea } from './layout';
+
+export const DiagramEditor = ({ diagram, selection, setSelection }: DiagramEditorProps) => {
```

Comment:
This should probably be named DiagramLayoutEngine or something like that

Reply:
Even better, it could propabbly be replaced by a hook like:

```
// In DiagramRepresentation.tsx
const { layout } = useLayout();
const previousLayoutedDiagram: Diagram | null = this.state.diagram;
const layoutedDiagram: Diagram = layout(previousLayoutedDiagram, convertedDiagram);
setState(prevState => ({ ...prevState, diagram: layoutedDiagram }));
```

Reply:
Done